src/main/kotlin/com/londogard/textgen/normalization/SoftMaxNormalization.kt

```diff
@@ -0,0 +1,18 @@
+package com.londogard.textgen.normalization
+
+import kotlin.math.pow
+
+class SoftMaxNormalization(override val temperature: Double) : Normalization {
+    override fun normalize(probabilities: List<Pair<Int, Double>>): List<Pair<Int, Double>> {
+        val probs = probabilities.map { (_, score) -> score }.toDoubleArray()
+
+        return probabilities.map { (index, _) -> index }.zip(normalize(probs).asList()) // TODO might be expensive to box
+    }
+
+    override fun normalize(probabilities: DoubleArray): DoubleArray {
+        for (i in probabilities.indices) probabilities[i] = probabilities[i].pow(1 / (1 - temperature))
```

Comment:
Maybe this looks more clean? Can probably be done in an even better way.

```suggestion
        val sum = probabilities.map { item -> probabilities[i].pow(1 / (1 - temperature)) }.sum()
```

Reply:
Remember, I'm mutating something here. Meaning that I'm actually changing `probabilities[i]`, hence this won't work unfortunately